lib/OpenQA/Worker/Job.pm

```diff
@@ -439,9 +441,8 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(m/,/, $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // UPLOAD_FROM_POOL);
```

Comment:
It would be better to use `$self->settings->{AUTOINST_UPLOAD_FROM_POOL}` here instead. Generally jobs without settings are accepted and in this case `$self->info->{settings}` would be `undef`. `$self->settings` is always supposed to return a HASH-ref.

Reply:
thanks for the hint ! done